egs2/slue-voxpopuli/slu1/local/prepare_nel_data.py

```diff
@@ -0,0 +1,169 @@
+                        done_processing = True
+
+                if update_words:
+                    _ = entity_alignments.setdefault(utt_id, [])
+                    for idx in range(curr_idx, curr_idx + tier_idx):
+                        label, start_time, end_time = all_word_alignments[utt_id][idx]
+                        if idx == curr_idx:
+                            entity_phrase = label
+                            entity_start_time = start_time
+                            entity_end_time = end_time
+                        else:
+                            if label != "":
+                                entity_phrase += " " + label
+                            entity_end_time = end_time
+                        all_word_alignments[utt_id][idx] = (
+                            "#" + label,
+                            start_time,
+                            end_time,
+                        )
+                    entity_alignments[utt_id].append(
+                        [entity_phrase, entity_start_time, entity_end_time]
+                    )
+                    curr_idx += len_phrase
+                    data_idx += 1
+                else:
+                    curr_idx += 1
+                if data_idx < len(gt_labels):
+                    wrd_lst, len_phrase = read_gt_sample(gt_labels, data_idx)
+            else:
+                curr_idx += 1
+            if curr_idx == len(all_word_alignments[utt_id]) and data_idx != len(
+                gt_labels
+            ):
+                import pdb
+
+                pdb.set_trace()
+                print(data_idx, len(gt_labels))
+                print(gt_labels)
+                print(text)
+                sys.exit("Process exited, possibly an issue with text processing.")
+
+
+def modify_word_alignments(dataset_obj, data_dir, data_split, extract_gt):
+    """
+    Modify the word alignments to mark entity phrases for evaluation
+    """
+    all_word_alignments = {}
+    entity_alignments = {}
+    for idx, _ in enumerate(tqdm(dataset_obj)):
+        gt_labels = []
+        utt_id = dataset_obj[idx]["id"]
+        all_word_alignments[utt_id] = []
+        wrd_durs = dataset_obj[idx]["word_timestamps"]
+        num_wrds = len(wrd_durs["word"])
+        nel_labels = dataset_obj[idx]["ne_timestamps"]
+        num_ne = len(nel_labels["ne_label"])
+        if num_ne != 0:
+            txt = dataset_obj[idx]["text"]
+            for ne_idx in range(num_ne):
+                start_id = nel_labels["start_char_idx"][ne_idx]
+                length = nel_labels["char_offset"][ne_idx]
+                phrase = txt[start_id : start_id + length]
+                gt_labels.append((phrase, start_id, length))
+        for wrd_idx in range(num_wrds):
+            word = wrd_durs["word"][wrd_idx]
+            start_sec = wrd_durs["start_sec"][wrd_idx]
+            end_sec = wrd_durs["end_sec"][wrd_idx]
+            all_word_alignments[utt_id].append([word, start_sec, end_sec])
```

Comment:
Replace line 143 with the following logic to appropriately handle any cases with an apostrophe

```
            if len(word) > 2 and word[-2:] == "'s": # handle apostrophe
                all_word_alignments[utt_id].append([word[:-2], start_sec, end_sec])
                all_word_alignments[utt_id].append([word[-2:], end_sec, end_sec])
            else:
                all_word_alignments[utt_id].append([word, start_sec, end_sec])

```

Reply:
Fixed